ark/phenotyping/som_utils.py

```diff
@@ -131,50 +111,174 @@ def compute_pixel_cluster_channel_avg(fovs, channels, base_dir, cluster_col,
         cluster_avgs = pd.concat([cluster_avgs, agg_results])
 
     # sum the counts and the channel sums
-    sum_count_totals = cluster_avgs.groupby(cluster_col)[channels + ['count']].sum().reset_index()
+    sum_count_totals = cluster_avgs.groupby(
+        pixel_cluster_col
+    )[channels + ['count']].sum().reset_index()
 
     # now compute the means using the count column
     sum_count_totals[channels] = sum_count_totals[channels].div(sum_count_totals['count'], axis=0)
 
+    # z-score the mean channel expressions
+    sum_count_totals[channels] = stats.zscore(sum_count_totals[channels])
```

Comment:
In reference to my other comment, this is where you are already z-scoring the data.